# After QPR3, c2 codecs crash on exynos5420. Revert back to omx codecs by reverting
#
# https://github.com/LineageOS/android_frameworks_av/commit/60ea1ce6e8f5e8b258e15a2f7eaf10b9563e8614
#
# logcat error message of c2 codec crashing
# 
# 09-20 12:26:54.632  3527  3783 I CCodecBufferChannel: [c2.android.vorbis.decoder#836] Created output block pool with allocatorID 16 => poolID 18 - OK
# 09-20 12:26:54.635  3527  3783 D CCodecBufferChannel: [c2.android.vorbis.decoder#836] Configured output block pool ids 18 => OK
# 09-20 12:26:54.637  3527  3784 E ion     : ioctl c0044901 failed with code -1: Invalid argument
# 09-20 12:26:54.638  3527  3783 E ion     : ioctl c0044901 failed with code -1: Invalid argument
# --------- beginning of crash
# 09-20 12:26:54.640  3527  3784 F libc    : Fatal signal 11 (SIGSEGV), code 2 (SEGV_ACCERR), fault addr 0xa8221000 in tid 3784 (NDK MediaCodec_), pid 3527 (system_server)
# 09-20 12:26:54.640  3502  3502 W AudioOut_D: type=1701 audit(0.0:206): auid=4294967295 uid=1041 gid=1005 ses=4294967295 subj=u:r:audioserver:s0 reason="memory violation" sig=11

diff --git a/media/libstagefright/omx/Android.bp b/media/libstagefright/omx/Android.bp
index 346f40f..92d48f4 100644
--- a/media/libstagefright/omx/Android.bp
+++ b/media/libstagefright/omx/Android.bp
@@ -187,7 +187,7 @@ cc_defaults {
     name: "libstagefright_softomx-defaults",
     // TODO (b/316432618) Software OMX codecs are no longer used, disable building them till
     // this code is removed completely.
-    enabled: false,
+    // enabled: false,
     vendor_available: true,
 
     cflags: [
diff --git a/services/mediacodec/Android.bp b/services/mediacodec/Android.bp
index 506b3bc..a2f17c2 100644
--- a/services/mediacodec/Android.bp
+++ b/services/mediacodec/Android.bp
@@ -95,6 +95,31 @@ cc_binary {
         "android.hidl.memory@1.0",
     ],
 
+    runtime_libs: [
+        "libstagefright_soft_aacdec",
+        "libstagefright_soft_aacenc",
+        "libstagefright_soft_amrdec",
+        "libstagefright_soft_amrnbenc",
+        "libstagefright_soft_amrwbenc",
+        "libstagefright_soft_avcdec",
+        "libstagefright_soft_avcenc",
+        "libstagefright_soft_flacdec",
+        "libstagefright_soft_flacenc",
+        "libstagefright_soft_g711dec",
+        "libstagefright_soft_gsmdec",
+        "libstagefright_soft_hevcdec",
+        "libstagefright_soft_mp3dec",
+        "libstagefright_soft_mpeg2dec",
+        "libstagefright_soft_mpeg4dec",
+        "libstagefright_soft_mpeg4enc",
+        "libstagefright_soft_opusdec",
+        "libstagefright_soft_rawdec",
+        "libstagefright_soft_vorbisdec",
+        "libstagefright_soft_vpxdec",
+        "libstagefright_soft_vpxenc",
+        "libstagefright_softomx_plugin",
+    ],
+
     // OMX interfaces force this to stay in 32-bit mode;
     compile_multilib: "32",
 
